code/_lighting/RobustLight2.dm

```diff
@@ -735,20 +721,18 @@ proc/get_moving_lights_stats()
 	//for fucks sake tobba - ZeWaka
 
 /turf/proc/RL_Init()
-	if (!fullbright && !loc:force_fullbright)
-		if(!src.RL_MulOverlay)
-			src.RL_MulOverlay = new /obj/overlay/tile_effect/lighting/mul(src)
+	if (fullbright || loc:force_fullbright)
+		src.RL_MulOverlay?.dispose()
+		src.RL_MulOverlay = null
+		src.RL_AddOverlay?.dispose()
+		src.RL_AddOverlay = null
```

Comment:
Provide note for why `dispose` is being used directly.

Reply:
qdel is a fair bit slow, and nothing should access the overlay except its own turf